### PR TITLE
Implemented Nag Dialog Visual & Code Improvements

### PR DESCRIPTION
**Pull request by IllianiBird**
This PR rewrites our current nag dialogs to be a bit more visually interesting - for the player - and more feature robust - for us. Essentially I moved us from the `AbstractNagDialog` class to `JDialog`. This gives us access to a far wide suite of tools, when writing nag dialogs. Specifically when it comes to how we present information to the user.

Something we noticed a lot was players were being presented with these nags, without knowing how to address them. The 'you have no campaign commander' one was the most common culprit of this. Using the old nag dialogs we were very, very limited in how we could layout information as we were unable to easily use `html` formatting. By using `JDialog` that ceases to be an issue.

All together that means we go from this...
image

...to this...
image

From an internals perspective, I moved the nag dialog triggers out of `CampaignGUI.java` and into its own nag controller. This helps reduce the size of `CampaignGUI.java` and keeps all the nag functionality in one place.

~~Finally, I removed the Unit Tests allocated to the various nag dialogs. While those tests were invaluable in helping me learn how to write Unit Tests, they weren't really necessary for nag dialogs. At best all they did was test the inner functionality of the various nags. At worst they broke any time one of the nags was modified.~~

### CODE REVIEW

**Comment by HammerGS** [comment]
I have to stay the Commander, there is no Commander is odd wording. 

I think for that it should be just Please be advised...

**Comment by IllianiCBT** [comment]
Done

**Comment by IllianiCBT** [comment]
This PR partially addresses #5522

File: MekHQ/resources/mekhq/resources/GUI.properties [hunk 1]
```diff
@@ -312,70 +312,115 @@ UnitIconDialog.title=Select Unit Icon
...
+  <br>\
+  <br><i>You should head to the Briefing Room and address this shortfall. Remember, Combat Teams\
+  \ assigned to the Auxiliary or Reserve roles do not count towards deployment levels. Nor do\
+  \ Combat Teams assigned to the Training role, unless the contract is Cadre Duty.</i>
+PrisonersNagDialog.text=%s, our forces are still holding prisoners of war. It is important to\
+  \ consider their status before advancing the day. Do you wish to continue without resolving this?\
+  <br>\
+  <br><i>You should head to the Personnel tab and set the filter to 'Prisoners'. Right-clicking on\
+  \ a prisoner will allow you to free, ransom, or execute them. If you are currently off-world, you\
+  \ can choose to jettison your prisoners</i>
```
Comment (NickAragua) [inline MekHQ/resources/mekhq/resources/GUI.properties:384]:
```suggestion
  \ can choose to jettison your prisoners.</i>
```

May also consider adding a :( to the end of the sentence, lol.

File: MekHQ/resources/mekhq/resources/GUI.properties [hunk 2]
```diff
@@ -312,70 +312,115 @@ UnitIconDialog.title=Select Unit Icon
...
+  \ exceeds available funds. Confirm your intent to advance the day without addressing this issue.\
+  <br>\
+  <br><i>Consider selling off unwanted units, or parts. If all else fails you might be able to take\
+  \ out another loan in the Finances tab./i>
+UnresolvedStratConContactsNagDialog.text=%s, reports confirm enemy activity within the Area of\
+  \ Operations. Please review the situation carefully and take any necessary action to ensure\
+  \ operational success.%s
+UnresolvedStratConContactsNagDialog.stratcon=<br>\
+  <br><i>We can pay 1 CVP to refuse to engage any contact marked as 'Critical.' It is free to\
+  \ refuse non-critical contacts.</i>
```
Comment (NickAragua) [inline MekHQ/resources/mekhq/resources/GUI.properties:407]:
```suggestion
  \ ignore non-critical contacts.</i>
```

**Review by NickAragua — CHANGES_REQUESTED** [review]
This is really cool and adds a lot more immersion and "new user help". Other than a couple minor text suggestions, the main issue I have with it is the removal of unit tests, which I think we should retain in some form to avoid regression bugs in the future.

**Comment by IllianiCBT** [comment]
-sacrifices a turkey in the hopes the tests pass-

**Comment by IllianiCBT** [comment]
Split the logic and gui elements of each nag into separate classes to more easily facilitate testing. As we test in an headless environment, here on GitHub, having gui elements in the testing causes no end of hassle. Separating the logic and testing that individually seemed like the best approach. It also has the added benefit of keeping everything nice and neatly self-contained.

**Comment by NickAragua** [comment]
Thanks for restoring the unit tests, good stuff.

**Pull request merged**